accepted/2020/form-factors.md

```diff
@@ -0,0 +1,206 @@
+- [Appendix](#appendix)
+  - [Lessons Learned from CoreRT](#lessons-learned-from-corert)
+
+## Dominant .NET Runtime Form Factors
+
+The form factors described in this section are the dominant ones that exist today.
+
+### Global and General Purpose
+
+The globally installed runtime has been the most successful runtime form factor since the inception of .NET (meaning, the 2001 introduction of the .NET Framework). It fits the needs of typical server-side web and cloud applications, and a number of other important .NET workloads. We think of this form factor as general purpose, covering scenarios (just at Microsoft) as diverse as the Visual Studio IDE, Azure and Office 365 services, and the C# compiler.
+
+This runtime form factor is relatively large (100s of MBs), supports all .NET features and is composed of many binaries (most small, some large). The runtime is often optimized for a specific platform (like Windows x64) using ahead-of-time (AOT) compilation. Typical applications are relatively small and contain only their own code and third-party dependencies. Applications depend on the presence of a specific version of the runtime (like .NET Framework 4.8 or .NET Core 3.1) for execution.
+
+### Self-Contained
+
+The globally installed runtime is not always desirable or even an option (e.g. due to admin privilege requirements). [Self-contained deployment](https://docs.microsoft.com/en-us/dotnet/core/deploying/#publish-self-contained) avoids the dependence on a global install by including the runtime with the application.
+
+The simplest and most compatible way to compose self-contained applications is to include all runtime binaries next to the application, which is an option that has been available since .NET Core 1.0. This approach produces applications with a large disk footprint that are composed of many files. These characteristics present real challenges and negative perception. To address those problems, we introduced an option as part of .NET Core 2.0 that removes unused files. This approach significantly reduced disk footprint, but did not resolve perception issues. We introduced a single-file self-extractor as part of .NET Core 3.0, which in theory helped with customer perception, but wasn't the single-file solution that most users wanted. We are incrementally improving this solution as part of .NET 5.0 to better align with what people expect (a single file executable that loads dependencies as in-memory resources where possible).
+
+The following feedback outlines customer-observed problems with the *single-file form* factor:
+
+- [SingleFilePublish is quite unusable](https://github.com/dotnet/corert/issues/7200#issuecomment-564778685)
+- [Current executing assembly location is wrong for single file published project](https://github.com/dotnet/runtime/issues/13531)
+- [PublishSingleFile with Windows Service doesn't work with config-file](https://github.com/dotnet/runtime/issues/32735)
```

Comment:
The above issues are very important and why we can't use single file/self-contained. Being able to define the extraction location yourself (relative to single file) would help a lot and then use SetDllDirectory (on windows) instead of changing current directory would perhaps be simple changes that could help previous dotnet framework devs.

We do single file publish to some degree by simply embedding all managed assemblies into exe, and copy native dlls to e. g. `x64` sub-directories. This keeps the root clean (next to exe). Hence, much better with regard to perception.